bouncerscript/script.py

```diff
@@ -62,55 +56,23 @@
 
 
 async def async_main(context):
-    context.task = get_task(context.config)
+    # perform schema validation for the corresponding type of task
+    validate_task_schema(context)
 
     # determine the task server and action
     context.server = get_task_server(context.task, context.config)
     context.action = get_task_action(context.task, context.config)
 
-    # perform schema validation for the corresponding type of task
-    validate_task_schema(context, context.action)
-
     # perform the appropriate behavior
-    await action_map[context.action](context)
-
-
-def usage():
-    print("Usage: {} CONFIG_FILE".format(sys.argv[0]), file=sys.stderr)
-    sys.exit(1)
-
-
-def craft_logging_config(context):
-    return {
-        'format': '%(asctime)s - %(name)s - %(levelname)s - %(message)s',
-        'level': logging.DEBUG if context.config.get('verbose') else logging.INFO
-    }
-
-
-def main(name=None, config_path=None):
-    if name not in (None, '__main__'):
-        return
-    context = Context()
-    context.config = dict()
-    if config_path is None:
-        if len(sys.argv) != 2:
-            usage()
-        config_path = sys.argv[1]
-    context.config.update(load_json(path=config_path))
-
-    logging.basicConfig(**craft_logging_config(context))
-    logging.getLogger('taskcluster').setLevel(logging.WARNING)
+    if action_map.get(context.action):
+        await action_map[context.action](context)
+    else:
+        log.critical("Unknown action: {}!".format(context.action))
+        sys.exit(3)
```

Comment:
Nit: How about `raise TaskValidationError("Unknown action: {}!".format(context.action))`? `scriptworker` deals with [exit statuses for you](https://github.com/mozilla-releng/scriptworker/blob/master/scriptworker/exceptions.py#L89).

Reply:
Sure, I've amended it.